Framework/Core/test/test_ASoA.cxx

```diff
@@ -286,14 +286,14 @@ TEST_CASE("TestJoinedTables")
     REQUIRE(7 == test.x() + test.y());
   }
 
-  auto tests2 = join(TestX{tableX}, TestY{tableY});
+  auto tests2 = join<o2::header::DataOrigin{"JOIN"}>(TestX{tableX}, TestY{tableY});
   static_assert(std::is_same_v<Test::table_t, decltype(tests2)>,
                 "Joined tables should have the same type, regardless how we construct them");
   for (auto& test : tests2) {
     REQUIRE(7 == test.x() + test.y());
   }
 
-  auto tests3 = join(TestX{tableX}, TestY{tableY}, TestZ{tableZ});
+  auto tests3 = join<o2::header::DataOrigin{"JOIN"}>(TestX{tableX}, TestY{tableY}, TestZ{tableZ});
```

Comment:
Same here. Is there any case where the origin is not "JOIN" actually?

Reply:
This function is generic, in principle, so it leaves the possibility to have different origin. However, partial specializations are not available for functions, so I cannot simply make an alias as with `Table`. Perhaps I can make this function a member of a helper struct, though.